scripts/gulp/tasks/test-projects.ts

```diff
@@ -17,16 +16,25 @@ type PackedPackages = Record<string, string>;
 
 const { paths } = config;
 
+// Clean up created files/folders on exit, even after exceptions
+// (will not catch SIGINT on windows)
+tmp.setGracefulCleanup();
+
+/** Shared packed packages between tests since they're not modified by any test */
+let packedPackages: PackedPackages;
+
 const log = (context: string) => (message: string) => {
   console.log();
   console.log('='.repeat(80));
   console.log(`${context} : ${message}`);
   console.log('='.repeat(80));
 };
 
-export const runIn = targetPath => cmd => sh(cmd, targetPath);
+const createTempDir = (prefix: string): string => {
+  return tmp.dirSync({ prefix, unsafeCleanup: true }).name;
```

Comment:
"Unsafe" means clean up even if there are still files in the directory (so perfectly safe in our case). Should probably have added that in a comment.